runtime/storage_test.go

```diff
@@ -1556,9 +1556,7 @@ func TestRuntimeStorageReferenceCast(t *testing.T) {
 		},
 	)
 
-	RequireError(t, err)
-
-	require.ErrorAs(t, err, &interpreter.ForceCastTypeMismatchError{})
+	require.NoError(t, err)
```

Comment:
Maybe add a test case that ensures that the type cannot be downcast with additional entitlements